src/Sudoku.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Maman 13 - Class Sudoku
+ *
+ * @author Noa Amir
+ * @version 2022a
+ */
+
+public class Sudoku {
+
+    // initialize final variables
+    private final int SIZE = 3;
+    private final boolean[] WHICH_NUMBERS_APPEAR = new boolean[SIZE*SIZE+1];
+
+    // instance variables
+    private Square3x3[][] sudoku;
+
+    // constructors
+
+    /**
+     * Construct a new Sudoku. Initialize all Square3x3 cells value to -1
+     */
+    public Sudoku() {
+        sudoku = new Square3x3[SIZE][SIZE];
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                sudoku[i][j] = new Square3x3();
+            }
+        }
+    }
+
+    /**
+     * Construct a new Sudoku. Copy values from a given Square3x3 array to sudoku cells
+     *
+     * @param array - Square3x3 array to copy the values from
+     */
+    public Sudoku(Square3x3[][] array) {
+        this(); // initialize sudoku cells value to -1
+
+        if (array != null) { // only if array parameter is not null, copy its values to the sudoku
+            sudoku = new Square3x3[SIZE][SIZE];
+            for (int i = 0; i < SIZE; i++) {
+                for (int j = 0; j < SIZE; j++) {
+                    sudoku[i][j] = new Square3x3(array[i][j]);
+                }
+            }
+        }
+    }
+
+    // methods
+
+    // Initialize WHICH_NUMBERS_APPEAR boolean array
+    // [0] item value with true
+    // [1] - [9] items value with false
+    private void initWhichNumbersAppear() {
+        WHICH_NUMBERS_APPEAR[0] = true;
+        for (int i = 1; i < WHICH_NUMBERS_APPEAR.length; i++) {
+            WHICH_NUMBERS_APPEAR[i] = false;
+        }
+    }
+
+    // Check if WHICH_NUMBERS_APPEAR boolean array has only true values
+    private boolean validateWhichNumbersAppear() {
```

Comment:
this method doesn't answer the question "is the array valid?".
Instead, it answers to a different question "are all numbers there?".
Hence, I would call it `doesAllNumbersAppear`.

Also, and this is a minor comment: consider replacing the word "appear" with `present/exist` (not just here).

Last, generally speaking - cool website for synonyms:
https://www.thesaurus.com/
It helps me to come up with better names frequently!